packages/transaction-controller/src/utils/batch.ts

```diff
@@ -573,19 +580,22 @@ async function processTransactionWithHook(
     };
   }
 
-  const { transactionMeta } = await addTransaction(
-    {
-      ...params,
-      from,
-    },
-    {
-      batchId,
-      disableGasBuffer: true,
-      networkClientId,
-      publishHook,
-      requireApproval: false,
-    },
-  );
+  const txMeta = { ...txBatchMeta, txParams: { ...params, from } };
```

Comment:
Minor, is it confusing to have this and `transactionMeta` below? Maybe we call this `transactionMetaForGasEstimates`?